src/flamenco/runtime/program/fd_bpf_loader_program.c

```diff
@@ -236,11 +237,13 @@ calculate_heap_cost( ulong heap_size, ulong heap_cost, int * err ) {
 int
 fd_deploy_program( fd_exec_instr_ctx_t * instr_ctx,
                    uchar const *         programdata,
-                   ulong                 programdata_size ) {
+                   ulong                 programdata_size,
+                   fd_valloc_t           valloc /* Majority of use cases this is a txn ctx spad under the hood; for native to bpf migration this is scratch backed since it's an epoch level event. */
```

Comment:
We also need to bound out the epoch level events too. Probably makes sense to do in a follow up PR

Reply:
Agreed.